src/openforms/js/components/admin/form_design/MissingTranslationsWarning.js

```diff
@@ -13,10 +13,28 @@ import {extractMissingComponentTranslations} from './MissingComponentTranslation
 const LABEL_MAPPING = jsonScriptToVar('label-mapping', {default: {}});
 const LANGUAGES = jsonScriptToVar('languages', {default: []});
 
-const extractMissingTranslations = (translations, tabName, fieldNames, fallbackFields) => {
+const extractMissingTranslations = (
+  translations,
+  tabName,
+  fieldNames,
+  fallbackFields,
+  optionalFields = []
+) => {
   const defaultLangCode = LANGUAGES[0][0];
   const languageCodeMapping = Object.fromEntries(LANGUAGES);
   let skipWarningsFor = [];
+  const numOptionalTranslations = {};
+
+  const numLanguages = Object.keys(translations).length;
+  // count the amount of optional field translations
+  for (const [languageCode, mapping] of Object.entries(translations)) {
+    for (const [key, translation] of Object.entries(mapping)) {
+      if (fieldNames && (fieldNames || []).includes(key) && optionalFields.includes(key)) {
```

Comment:
```suggestion
      if (fieldNames && fieldNames.includes(key) && optionalFields.includes(key)) {
```
the `if fieldNames` already establishes that the variable is not `undefined`, `null` or otherwise falsy.